### PR TITLE
[DDW-844] Fix redeem ITN rewards wallet dropdown / introduce disabled state to the Select component

### PR DESCRIPTION
**Pull request by marcin-mazurek**
This PR resolves issues introduced in PR https://github.com/input-output-hk/daedalus/pull/2816 and then reverted in PR https://github.com/input-output-hk/daedalus/pull/2891.

## Before

Screenshot 2022-01-25 at 20 18 22

Screenshot 2022-01-25 at 20 18 33

## After

Screenshot 2022-01-25 at 20 02 12

Screenshot 2022-01-25 at 19 58 33

Screenshot 2022-03-04 at 23 36 15

Screenshot 2022-03-04 at 23 39 02

## Test Scenarios

```Gherkin
Scenario 1: CSS styling has no outline but full background
Given I an on redeem ITN screen
When I observe the Redeem rewards to dropdown css styling
Then I observe the css styling has no outline
And the css styling has full background

Scenario 2: Dropdown remains enabled after phrase removed
Given I am on redeem ITN screen
And I have pasted valid wallet recovery phrase into text area
And Redeem rewards to dropdown is enabled
When I remove one or more phrase from the text area
Then the Redeem rewards to dropdown remains enabled
And the dropdown is functioning as expected

Scenario 3: Dropdown selection remains selected after changing valid recovery phrase to invalid
Given I am on redeem ITN screen
And I have pasted valid wallet recovery phrase into text area
And Redeem rewards to dropdown is enabled
And I have selected an option from the Redeem rewards to dropdown
When I remove one or more phrase from the text area
Or When I change the pasted valid recovery phrase to invalid recovery phrase
Then the Redeem rewards to dropdown remains enabled
And the previously selected dropdown option remains selected
```

- [Slack QA thread](https://github.com/input-output-hk/daedalus/pull/2907)
- [ ] Test

---

### Basics
- [x] PR assigned to the PR author(s) 
- [x] `input-output-hk/daedalus-dev` and `input-output-hk/daedalus-qa` assigned as PR reviewers
- [x] If there are UI changes, Alexander Rukin assigned as an additional reviewer
- [x] PR has appropriate labels (`release-vNext`, `feature`/`bug`/`chore`, `WIP`)
- [x] PR link is added to a Jira ticket
- [x] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [x] PR has a good description that summarizes all changes
- [x] PR contains screenshots (in case of UI changes)
- [x] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [x] There are no missing translations (running `yarn manage:translations` produces no changes)
- [x] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [x] Japanese text changes are proofread and approved (Junko Oda)
- [x] Storybook works and no stories are broken (`yarn storybook`)
- [x] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [x] Important parts of the code are properly commented and documented
- [x] Code is properly typed with flow
- [x] React components are split-up enough to avoid unnecessary re-renderings
- [x] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by alexander-rukin** [comment]
seems like it breaks send screen again 😱 

image

**Comment by marcin-mazurek** [comment]
@alexander-rukin please have a look now, the PR was marked as WIP when you checked it :)

File: source/renderer/app/i18n/locales/defaultMessages.json [hunk 1]
```diff
@@ -4635,7 +4635,7 @@
         }
```
Comment (marcin-mazurek) [inline source/renderer/app/i18n/locales/defaultMessages.json:4635]: Someone forgot to run `yarn manage:translations` and it was merged to develop.

File: source/renderer/app/i18n/locales/en-US.json [hunk 2]
```diff
@@ -1298,4 +1298,4 @@
   "wallet.transferFunds.dialog2.total.label": "Total",
```
Comment (marcin-mazurek) [inline source/renderer/app/i18n/locales/en-US.json:1298]: Someone forgot to run `yarn manage:translations` and it was merged to develop.

File: source/renderer/app/i18n/locales/ja-JP.json [hunk 3]
```diff
@@ -1298,4 +1298,4 @@
   "wallet.transferFunds.dialog2.total.label": "合計",
```
Comment (marcin-mazurek) [inline source/renderer/app/i18n/locales/ja-JP.json:1298]: Someone forgot to run `yarn manage:translations` and it was merged to develop.

**Review by shawnbusuttil — APPROVED** [review]

**Review by lucas-barros — APPROVED** [review]

File: source/renderer/app/components/staking/redeem-itn-rewards/Step1ConfigurationDialog.tsx [hunk 4]
```diff
@@ -276,11 +295,11 @@ class Step1ConfigurationDialog extends Component<Props> {
...
     const walletId = get(wallet, 'id', null);
-    const validRecoveryPhase = recoveryPhraseField.isValid;
-    const walletsDropdownClasses = classnames([
-      styles.walletsDropdown,
-      !validRecoveryPhase ? styles.disabled : null,
-    ]);
+    const walletsDropdownDisabled = !(
+      recoveryPhraseField.isValid ||
+      this.state.wasRecoveryPhraseValidAtLeastOnce
+    );
```
Comment (szymonmaslowski) [inline source/renderer/app/components/staking/redeem-itn-rewards/Step1ConfigurationDialog.tsx:301]:
```suggestion
    const walletsDropdownDisabled = !recoveryPhraseField.isValid && !this.state.wasRecoveryPhraseValidAtLeastOnce;
```

File: source/renderer/app/i18n/locales/defaultMessages.json (hunk 1 above)
Comment (szymonmaslowski) [reply source/renderer/app/i18n/locales/defaultMessages.json:4635]: It means that our CI didn't catch that :/

Comment (marcin-mazurek) [reply source/renderer/app/i18n/locales/defaultMessages.json:4635]: It must've been before my CI fix was merged

File: source/renderer/app/components/staking/redeem-itn-rewards/Step1ConfigurationDialog.tsx (hunk 4 above)
Comment (marcin-mazurek) [reply source/renderer/app/components/staking/redeem-itn-rewards/Step1ConfigurationDialog.tsx:301]: I prefer single negation actually, so first you read `recoveryPhraseField.isValid ||      this.state.wasRecoveryPhraseValidAtLeastOnce` and figure out that these are conditions under which the dropdown should be **enabled**, and later we just negate the whole expression to turn it into **disabled**

Comment (szymonmaslowski) [reply source/renderer/app/components/staking/redeem-itn-rewards/Step1ConfigurationDialog.tsx:301]: For me the other way around is simpler to read and understand the expected values of each "ingredient" if the negation happens for each of them. When negating once at the root I need to deduct expected value of each variable. This is however a subjective thing and when I started to search the internet for any good practice regarding this subject I found articles against my suggestion, so I cannot convince you :)

However I have another suggestion that I think both of us will prefer:
```suggestion
    const walletsDropdownEnabled = recoveryPhraseField.isValid ||
      this.state.wasRecoveryPhraseValidAtLeastOnce;
```
What do you think? :)

**Review by dmitrii-gaico — APPROVED** [review]

**Review by miorsufianiohk — APPROVED** [review]
LGTM. Great work @marcin-mazurek 👍 . Tested on 21172

**Pull request merged**